Code/GraphMol/FMCS/testFMCS_Unit.cpp

```diff
@@ -2473,6 +2473,53 @@ void testAtomCompareCompleteRingsOnly() {
   BOOST_LOG(rdInfoLog) << "\tdone" << std::endl;
 }
 
+void testGitHub4498() {
+  BOOST_LOG(rdInfoLog) << "-------------------------------------" << std::endl;
+  BOOST_LOG(rdInfoLog) << "GitHub Issue #4498: FindMCS may leave mols with "
+                          "fake (empty) ring info"
+                       << std::endl;
+  {
+    // Test with rings
+    std::vector<ROMOL_SPTR> mols = {"NC1=CC(N)=C(N)C=C1"_smiles,
```

Comment:
huh, I didn't realize you can safely initialize a boost::shared_ptr from a std::unique_ptr, but it apparently works since boost 1.53

That's useful!

Reply:
Same here. I just noticed when I copied part of the code of the test above to get mine started.